java/src/org/openqa/selenium/chromium/ChromiumNetworkConditions.java

```diff
@@ -0,0 +1,107 @@
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.openqa.selenium.chromium;
+
+import java.time.Duration;
+
+/**
+ * Provides manipulation of getting and setting network conditions from Chromium.
+ *
+ */
+public class ChromiumNetworkConditions {
+  private Boolean offline = false;
+  private Duration latency = Duration.ZERO;
+  private Number downloadThroughput = -1;
+  private Number uploadThroughput = -1;
+
+  public static final String OFFLINE = "offline";
+  public static final String LATENCY = "latency";
+  public static final String DOWNLOAD_THROUGHPUT = "download_throughput";
+  public static final String UPLOAD_THROUGHPUT = "upload_throughput";
+
+  /**
+   *
+   * @return whether network is simulated to be offline.
+   */
+  public Boolean getOffline() {
+    return offline;
+  }
+
+  /**
+   * Whether the network is set to offline. Defaults to false.
+   *
+   * @param offline when set to true, network is simulated to be offline.
+   */
+  public void setOffline(Boolean offline) {
+    this.offline = offline;
+  }
+
+  /**
+   *  The current simulated latency of the connection.
+   *
+   * @return amount of latency, typically a Duration of milliseconds.
+   */
+  public Duration getLatency() {
+    return latency;
+  }
+
+  /**
+   *  Sets the simulated latency of the connection.
+   *
+   * @param latency amount of latency, typically a Duration of millisceonds.
+   */
+  public void setLatency(Duration latency) {
+    this.latency = latency;
+  }
+
+  /**
+   * The current throughput of the network connection in kb/second for downloading.
+   *
+   * @return the current download throughput in kb/second.
+   */
+  public Number getDownloadThroughput() {
+    return downloadThroughput;
+  }
+
+  /**
+   * ets the throughput of the network connection in kb/second for downloading.
+   *
+   * @param downloadThroughput thoughput in kb/second
+   */
+  public void setDownloadThroughput(Number downloadThroughput) {
+    this.downloadThroughput = downloadThroughput;
+  }
+
+  /**
+   * The current throughput of the network connection in kb/second for uploading.
+   *
+   * @return the current upload throughput in kb/second.
+   */
+  public Number getUploadThroughput() {
+    return uploadThroughput;
+  }
+
+  /**
+   * ets the throughput of the network connection in kb/second for uploading.
+   *
+   * @param uploadThroughput thoughput in kb/second
+   */
+  public void setUploadThroughput(Number uploadThroughput) {
+    this.uploadThroughput = uploadThroughput;
+  }
+}
```

Comment:
You may need a `private static ChromiumNetworkConditions fromJson(JsonInput input)` here, but if the setters take primitives, I think it'll work without it.

Reply:
I mean, the tests were passing even with `Number`, so the right thing has to be happening?